tests/functional/olp-cpp-sdk-dataservice-read/VersionedLayerClientGetPartitionTest.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+#include <olp/core/client/OlpClientSettings.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/dataservice/read/VersionedLayerClient.h>
+// clang-format off
+#include "generated/serializer/PartitionsSerializer.h"
+#include "generated/serializer/JsonSerializer.h"
+// clang-format on
+#include "SetupMockServer.h"
+#include "ApiDefaultResponses.h"
+#include "MockServerHelper.h"
+#include "ReadDefaultResponses.h"
+#include "Utils.h"
+
+namespace {
+
+const auto kTestHrn = "hrn:here:data::olp-here-test:hereos-internal-test";
+const auto kLayer = "testlayer";
+const auto kVersion = 44;
+constexpr auto kWaitTimeout = std::chrono::seconds(10);
+const auto kPartitionsResponsePath =
+    "/query/v1/catalogs/hrn:here:data::olp-here-test:hereos-internal-test/"
+    "layers/testlayer/partitions";
+
+class VersionedLayerClientGetPartitionTest : public ::testing::Test {
```

Comment:
Its cool to make 1 file per api, but name becomes too complicated.
Maybe we can create 1 test class/alias for all versioned client tests, so test names are readable, e.g. TEST_F(VersionedLayerClient, GetPartitionsAsync)